toolz/utils.py

```diff
@@ -6,4 +6,11 @@ def raises(err, lamda):
         return True
 
 
-no_default = '__no__default__'
+@object.__new__
+class no_default(object):
+    def __new__(self):
+        raise TypeError("cannot create 'no_default' instances")
+
+    def __str__(self):
+        return '<object no_default>'
+    __repr__ = __str__
```

Comment:
Yes, that could happen.  My approach here would be to just make the string more strange.

Fancy methods also invite time bombs, but they're harder to reason about.  For example this object apparently isn't picklable.  

```python
In [1]: from toolz.utils import no_default

In [2]: no_default
Out[2]: <object no_default>

In [3]: import pickle

In [4]: pickle.dumps(no_default)
PicklingError: Can't pickle <class 'toolz.utils.no_default'>: it's not the same object as toolz.utils.no_default
```

Even if you were to fix this particular error (perhaps by making a `__getstate__` method) this is still an open Pandoras box.  It didn't take me long to find this particular issue.  I'm confident that there are more surprises with how an object like this would interact with the rest of the ecosystem.  Custom tricks like this are hard for me to reason about; my intuition for them isn't as built up as it is for strings.

I'm more comfortable with a very slowly ticking time bomb (that we can make arbitrarily slow) than with a Pandora's box.

Also, I avoid the term `class` within `toolz` to an arguably irrational extent.